src/lib/permissions/PermissionsAPI.web.js

```diff
@@ -1,17 +1,112 @@
+        return this._getClipboardPermissionStatus()
+
+      default:
+        return false
+    }
+  }
+
+  /*
+   * Private method used to verify if Browser's PermissionAPI is available and get permission status by provided kind
+   * @private
+   * @param {string} kind
+   * @return (undefined - Browser's PermissionAPI is not available) or (Promise<string> - one of PermissionStatusesEnum)
+   */
+  async _getStatusByBrowsersPermissionAPI(kind) {
+    // get Browser's PermissionAPI object
+    const permissionsApi = navigator.permissions
+
+    // check if Browser's PermissionsAPI is available in current browser
+    if (permissionsApi) {
+      // if Browser's PermissionsAPI i available then fetch permission status by received kind
+      // query method will return PermissionStatus object
+      const permissionStatusObj = await permissionsApi.query({ name: kind })
+
+      // fetch and return permission status from 'PermissionStatus.state' field
+      return permissionStatusObj.state
+    }
+  }
+
+  /*
+   * Private method used to get camera permission status
+   * @private
+   * @return Promise<string> - one of PermissionStatusesEnum
+   */
+  async _getCameraPermissionStatus() {
+    // get camera permission status by Browser's PermissionsAPI
+    const browserPermissionStatus = await this._getStatusByBrowsersPermissionAPI('camera')
+
+    // check if value exists
+    if (browserPermissionStatus) {
+      // return if exists
+      return browserPermissionStatus
+    }
+
+    // get camera status by requesting a video stream manually
+    return this._getCameraPermissionStatusByRequestingStream()
   }
 
-  async getClipboardPermissionStatus() {
+  /*
+   * Private method used to get camera permission status by manually request video stream
+   * Used when Browser's PermissionsAPI is not available in current browser
+   * @private
+   * @return Promise<string> - one of [PermissionStatusesEnum.DENIED, PermissionStatusesEnum.GRANTED]
+   */
+  async _getCameraPermissionStatusByRequestingStream() {
+    // fetching the getUserMedia method from navigator
+    const getUserMedia = navigator.mediaDevices.getUserMedia
+
+    // verify if getUserMedia is available
+    if (!getUserMedia) {
+      // make log - getUserMedia is not supported
+      log.warn('getUserMedia() is not supported by this browser')
+
+      // return denied as local video stream is not supported in this browser
+      return PermissionStatusesEnum.DENIED
+    }
+
+    try {
+      // requesting video stream to verify its available
+      await getUserMedia()
+    } catch (e) {
+      // make log of failed video stream request
+      log.warn('getUserMedia failed:', e.message, e)
+
+      // return denied for failure cases
+      return PermissionStatusesEnum.DENIED
+    }
+
+    // if the code reached this lines - it mean that video stream request is successful and permission is allowed
+    // return granted status
+    return PermissionStatusesEnum.GRANTED
```

Comment:
the final result should be true/false.
you should use those statuses only internally, to keep the detection flow i've listed in the comments